R/event.xyplot.hydromad.R

```diff
@@ -48,3 +99,13 @@ event.xyplot.hydromad.runlist <-
     foo$call <- sys.call(sys.parent())
     foo
   }
+
+#' @rdname event.xyplot.hydromad
+#' @export
+event.xyplot.hydromad.runlist <- event.xyplot.hydromad
+
+panel.groups.funs <- function(x, y, ...) {
+  panel.xyplot(x, y, ...)
+  if (!requireNamespace("mgcv")) stop("package mgcv is required for event.xyplot")
+  panel.smoother(x, y, y ~ s(x), method = "gam", ...)
+}
```

Comment:
I'm not keen on this solution given it defeats the purpose of having an in-line function definition, but happy to live with it.